### PR TITLE
Oceanwater 955 delivery command unification

### PR DESCRIPTION
**Pull request by anthonykoutroulis**
## Linked Issues:
| EPIC ⚡ | [OCEANWATER-933](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-933) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-955](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-955) |

### Summary of Changes
* Remove goal, feedback, and result components of Deliver action.
* Rename action to TaskDeliverSample.
* Move action file to owl_msgs package.
* Update references to be consistent with new name.

### Test
* This branch must be tested with the corresponding branch in `ow_autonomy`. 
* Do a clean build.  This itself verifies most of renaming and moving.
* Start any simulator world
* Run `task_deliver_sample_action_client.py` and see that the arm performs the deliver movement normally.

### CODE REVIEW

File: ow_lander/scripts/activity_deliver_sample.py [hunk 1]
```diff
@@ -18,9 +18,9 @@ def arg_parsing(req):
   """
   if req.use_defaults:
     # Default trenching values
-    x_delivery = 0.55
-    y_delivery = -0.3
-    z_delivery = 0.82  # was .78
+    x_delivery = rospy.get_param("deliver/x_delivery")
+    y_delivery = rospy.get_param("deliver/y_delivery")
+    z_delivery = rospy.get_param("deliver/z_delivery")  # was .78
```
Comment (kmdalal) [inline ow_lander/scripts/activity_deliver_sample.py:23]: Could delete the "was .78" comment since it is old and probably obsolete by now.

**Comment by AstroStucky** [comment]
The file `activity_deliver_sample.py` is actually and old ROS Service implementation of deliver, and is not used by the ROS Action. Apologies for the confusion, I have been meaning to remove these old and unused implementations for a month now. 

The delivery action is actually implemented inside of `trajectory_planner.py` along with the other action trajecotires. You'll fine it under the method `deliver_sample()` method. It looks like the implementation has gone through a lot of changes since the ROS Service days.

I dug a little deeper and it turns out the current implementation kind of already is parameterizes the deliver locations in `lander.srdf` since `deliver_sample()` implements deliver by calling a series of joint angles. However, those are joint angles, not a position in space, and modifying a delivery location from them is impractical. I recall the ROS Service Deliver gave us difficulties in the past, but I don't know the exact reason why it was switch from using a pose target to using joint angle targets.

**Comment by AstroStucky** [comment]
I just noticed this was a draft, so sorry if my comments are premature.

**Comment by anthonykoutroulis** [comment]
> I just noticed this was a draft, so sorry if my comments are premature.

Nope, all very helpful! You've both addressed all the questions I had while it was still a draft. I'll clean this up and ping you for proper review

**Comment by anthonykoutroulis** [comment]
I've removed the parameterize changes as OCEANWATER-954 is now a little more complicated than originally assumed, and moved from this release.

**Review by kmdalal — APPROVED** [review]
Looks good!  I think one review is sufficient, so removed @AstroStucky .

**Pull request merged**